src/main/java/org/walkmod/WalkModFacade.java

```diff
@@ -219,6 +223,58 @@ public WalkModFacade() {
 		this(false, true, false);
 	}
 
+
+	/**
+	 * Initalizes a Walkmod service
+	 *
+	 * @param walkmodCfg
+	 *            Walkmod configuration file. If null, file named 'walkmod.xml'
+	 *            is searched at the root.
+	 * @param options
+	 *            Map of option. See {@link Options} and {@link OptionsBuilder}
+	 *            for available options and default values.
+	 * @param configurationProvider
+	 *            Configuration provider responsible for the resolution of
+	 *            plugins (used to use custom classloading strategies).
+	 *            If null Ivy is used.
+	 */
+	public WalkModFacade(File walkmodCfg, Map<String,Object> options, ConfigurationProvider configurationProvider) {

```

Comment:
Also, I preffer to have a single method that supports nulls (with good JavaDocs) than having several methods. I know it's a personal thing and not everyone agrees. So If you prefare otherwise just let me know.